test-dts/shallowMount.d-test.ts

```diff
@@ -21,12 +21,12 @@ let wrapper = shallowMount(AppWithDefine, {
 // vm is properly typed
 expectType<string>(wrapper.vm.a)
 
-// can not receive extra props
-expectError(
-  shallowMount(AppWithDefine, {
-    props: { a: 'Hello', c: 2 }
-  })
-)
+// // can not receive extra props
```

Comment:
How come we don't need these anymore? Shouldn't these continue to error out if we infer the props?

Reply:
This will make the passing props a bit more relaxed, is also the same behaviour as [h#vue-next](https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/src/h.ts#L127)

Reply:
I think it would be better to uncomment the test and remove the `expectError` then, with a comment explaining that extra props is allowed as it is for `h()`. I don't like having commented code, because we will forget why ^^. And if the tests check that extra props are allowed, we are sure to avoid regressions in the future.